### PR TITLE
Fixes #262 - Integration testing more stable in maven.

### LINKED ISSUE #262: mvn clean test failing
In the async-support branch a `mvn clean test` fails, but `mvn clean install` passed

### PR DESCRIPTION
**Pull request by joakime**
+ Forces maven-failsafe-plugin in parent (with version)
+ Moves jetty9-compact-base tests back into src/test/java
+ Renames integration tests from *Test.java to *IT.java to
  better conform to new recommendations from maven-failsafe-plugin
  (convention over configuration!)

### CODE REVIEW

**Comment by joakime** [comment]
This PR isn't sane (yet), a followup commit (to correct checkstyle warnings) is due for this PR.

**Comment by joakime** [comment]
Google Style Guide is incompatible with maven-failsafe-plugin.

> /home/joakim/code/jetty/appengine-java-vm-runtime/jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeJettyAuthIT.java:30: error: Abbreviation in name must contain no more than '1' capital letters.

We'll need an exception.

**Comment by joakime** [comment]
@aozarov do you want me to alter the Checkstyle xml?

File: jetty9-compat-base/src/test/java/com/google/apphosting/runtime/jetty9/SessionManagerIT.java [hunk 1]
```diff
@@ -68,12 +62,18 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpSession;
 
+import static org.easymock.EasyMock.createMock;
```
Comment (aozarov) [inline jetty9-compat-base/src/test/java/com/google/apphosting/runtime/jetty9/SessionManagerIT.java]: Not sure why this was changed. static imports should be at the top according to the [Google style guide](https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing)

File: jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/AsyncServletIT.java [hunk 2]
```diff
@@ -22,17 +22,17 @@
...
-import org.apache.commons.httpclient.methods.RequestEntity;
-
 import java.io.IOException;
 import java.io.OutputStream;
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
 import java.util.List;
 
-public class AsyncServletTest extends VmRuntimeTestBase {
+import org.apache.commons.httpclient.HttpClient;
```
Comment (aozarov) [inline jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/AsyncServletIT.java]: Also this import should go before the java.* imports

File: jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/JettyRunner.java [hunk 3]
```diff
@@ -21,6 +21,13 @@
 
 import com.google.apphosting.vmruntime.VmRuntimeFileLogHandler;
 
+import java.io.File;
```
Comment (aozarov) [inline jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/JettyRunner.java]: goes bellow the org.* imports (as before).

File: jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/LoggingIT.java [hunk 4]
```diff
@@ -21,16 +21,16 @@
...
-import org.apache.commons.httpclient.methods.GetMethod;
-
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.InputStreamReader;
 import java.nio.charset.StandardCharsets;
 
-public class LoggingTest extends VmRuntimeTestBase {
+import org.apache.commons.httpclient.HttpClient;
```
Comment (aozarov) [inline jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/LoggingIT.java]: fix imports.

File: jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeJettyKitchenSink2IT.java [hunk 5]
```diff
@@ -25,22 +25,22 @@
 import com.google.appengine.api.taskqueue.TaskQueuePb.TaskQueueServiceError.ErrorCode;
 import com.google.apphosting.api.ApiBasePb.VoidProto;
 import com.google.apphosting.api.ApiProxy;
-//import com.google.apphosting.datastore.DatastoreV3Pb.Transaction;
 import com.google.apphosting.api.DatastorePb.Transaction;
 
+import java.net.HttpURLConnection;
```
Comment (aozarov) [inline jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeJettyKitchenSink2IT.java]: fix imports and would be nice to remove the commented one.

File: jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeJettyKitchenSinkIT.java [hunk 6]
```diff
@@ -32,11 +29,13 @@
 
 import javax.servlet.http.HttpServletResponse;
 
+import org.apache.commons.httpclient.HttpClient;
```
Comment (aozarov) [inline jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeJettyKitchenSinkIT.java]: fix imports.

File: jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeJettySessionIT.java [hunk 7]
```diff
@@ -28,23 +28,23 @@
...
-import org.apache.commons.httpclient.methods.GetMethod;
-
 import java.io.IOException;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.util.Arrays;
 
 import javax.servlet.http.HttpServletResponse;
 
+import org.apache.commons.httpclient.HttpClient;
```
Comment (aozarov) [inline jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeJettySessionIT.java]: fix imports.

File: jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeTestBase.java [hunk 8]
```diff
@@ -37,9 +35,11 @@
 
 import javax.servlet.http.HttpServletResponse;
 
+import org.junit.Ignore;
```
Comment (aozarov) [inline jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeTestBase.java]: fix imports.

File: jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContextTest.java [hunk 9]
```diff
@@ -40,9 +32,16 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import org.eclipse.jetty.server.LocalConnector;
```
Comment (aozarov) [inline jetty9-compat-base/src/test/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContextTest.java]: fix imports.

**Comment by aozarov** [comment]
> @aozarov do you want me to alter the Checkstyle xml?

Please do. it should be fine to have an IT suffix for integration tests and exempt it from checkstyle.

**Comment by joakime** [comment]
@aozarov why didn't the `google-java-format-1.0-all-deps.jar` fix these import issues?

**Comment by joakime** [comment]
Ah, [import management is currently broken](https://github.com/google/google-java-format/issues/42) in the `google-java-format-1.0-all-deps.jar`

Also, looks like there's a `googleformat-maven-plugin` in existence (maintained by @talios)

File: checkstyle/checkstyle-suppressions.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+    "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+    "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+    <suppress files="[a-zA-Z0-9]*IT.java"         checks="AbbreviationAsWordInName"/>
```
Comment (aozarov) [inline checkstyle/checkstyle-suppressions.xml]: Nit, should it be prefixed with  `[/\\]test[/\\]src[/\\]java[/\\]` (also for the other IT, Test checks)?

Comment (joakime) [inline checkstyle/checkstyle-suppressions.xml]: I would vote for the simpler form, as maven is free to copy/move files around into the `/target/` (aka `${project.build.directory}`) directory, which checkstyle also looks inside of.

Comment (aozarov) [inline checkstyle/checkstyle-suppressions.xml]: I can live with it though I wish we could avoid being lenient where were we shouldn't.

**Pull request merged**